### PR TITLE
Replaces uses of `cirq_ft` with `qualtran` throughout the library

### PR DESCRIPTION
**Pull request by tanujkhattar**
Part of [Qualtran & Cirq-FT Integration](https://docs.google.com/document/d/1miypjcCNt6JV_qAypUVPy5eH3EmKadLqeZomA5QoWK0/edit?usp=sharing&resourcekey=0-ciQ8jPfQmK8gtey2lNKesw)

After this PR, only the jupyter notebook generation code and cirq-interop code contains references to `cirq_ft`. These will be cleared in follow-up PRs.

### CODE REVIEW

File: qualtran/bloqs/chemistry/thc.py [hunk 1]
```diff
@@ -33,8 +31,10 @@
 from qualtran.bloqs.basic_gates import Hadamard, Ry, Toffoli, XGate
 from qualtran.bloqs.multi_control_multi_target_pauli import MultiControlPauli
 from qualtran.bloqs.on_each import OnEach
+from qualtran.bloqs.select_swap_qrom import SelectSwapQROM
```
Comment (fdmalone) [inline qualtran/bloqs/chemistry/thc.py:34]: There's a CirqGateAsBloq in the body, is this problematic?

**Review by fdmalone — CHANGES_REQUESTED** [review]
I'm blocking this myself just to check if the THC notebook renders nicely.

File: qualtran/bloqs/chemistry/thc_notebook_utils.py [hunk 2]
```diff
@@ -79,27 +80,11 @@ def generalize(bloq):
         return attrs.evolve(bloq, angle=phi_sym)
     if isinstance(bloq, And):
         return attrs.evolve(bloq, cv1=and_cv0, cv2=and_cv1)
+    if isinstance(bloq, SelectSwapQROM):
```
Comment (fdmalone) [inline qualtran/bloqs/chemistry/thc_notebook_utils.py:83]: This throws and error: AttributeError: 'QROM' object has no attribute 'gate', I wonder if this is necessary following #428

File: qualtran/bloqs/chemistry/thc.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/bloqs/chemistry/thc.py:34]: updated to not use `CirqGateAsBloq`

File: qualtran/bloqs/chemistry/thc_notebook_utils.py (hunk 2 above)
Comment (tanujkhattar) [reply qualtran/bloqs/chemistry/thc_notebook_utils.py:83]: fixed the error.

**Review by fdmalone — APPROVED** [review]

**Comment by fdmalone** [comment]
LGTM

**Pull request merged**